File: src/main/java/com/oz/office_tastezip/domain/user/repository/UserRepositoryCustom.java
package com.oz.office_tastezip.domain.user.repository;

import com.oz.office_tastezip.domain.user.User;

import java.util.Optional;

public interface UserRepositoryCustom {
    Optional<User> findByUserUUID(String uuid);

    int countByEmail(String email);

    void deleteByUserUUID(String uuid);
}

File: src/main/java/com/oz/office_tastezip/domain/user/repository/UserRepositoryCustom.kt
package com.oz.office_tastezip.domain.user.repository

import com.oz.office_tastezip.domain.user.User
import com.oz.office_tastezip.domain.user.dto.UserRequestDto
import java.util.UUID

interface UserRepositoryCustom {
    fun findByUserUUID(uuid: String): User?
    fun countByEmail(email: String): Int
    fun deleteByUserUUID(uuid: String)
    fun updateByUserUUID(userUpdateRequest: UserRequestDto.UserUpdateRequest)
    fun updateLastLoginAtByUserUUID(uuid: String)
    fun resetPassword(uuid : UUID, passwordHash: String)
}
